scarf/readers.py

```diff
@@ -382,6 +392,86 @@ def _read_dataset(self, key: Optional[str] = None):
             vals = None
         return vals
 
+    def read_header(self) -> pl.DataFrame:
+        header = pl.read_csv(
+            self.matFn,
+            skip_rows=2,
+            separator=self.sep,
+            has_header=False,
+            n_rows=1,
+            new_columns=["nFeatures", "nCells", "nCounts"],
+        )
+        return header
+
+    def process_batch(self, dfs: pl.DataFrame, filtering_cutoff: int) -> List:
+        """Returns a list of valid barcodes after filtering out background barcodes for a given batch.
+
+        Args:
+            dfs: A Polar DataFrame containing a chunk of data from the MTX file.
+            filtering_cutoff: The cutoff value for filtering out background barcodes
+        """
+        dfs_ = dfs.group_by('barcode').agg(pl.sum('count'))
+        dfs_ = dfs_.filter(pl.col('count') > filtering_cutoff)
+        return np.sort(dfs_['barcode'])
+
+    def _get_valid_barcodes(
+        self, filtering_cutoff: int,
+        batch_size: int = int(10e4),
+        lines_in_mem: int = int(10e6)
+    ) -> np.ndarray:
+        """Returns a list of valid barcodes after filtering out background barcodes.
+
+        Args:
+            filtering_cutoff: The cutoff value for filtering out background barcodes.
+            batch_size: The number of barcodes to process in each batch.
+            lines_in_mem: The number of lines to read into memory
+        """
+        test_counter = 0
+        matrixIO = pl.scan_csv(
+            self.matFn, skip_rows=3, separator=self.sep, has_header=False,
```

Comment:
.mtx can have several lines of headers (all starting with %), so it could potentially be brittle with a fixed `skip_rows`. 
https://math.nist.gov/MatrixMarket/formats.html

Maybe you can set the `comment_prefix` argument to `%` and test if it will work without a hard coded `skip_rows`?

Reply:
Thanks for pointing this out! I will edit it. 😊
I was following [this](https://kb.10xgenomics.com/hc/en-us/articles/115000794686-How-is-the-MEX-format-used-for-the-gene-barcode-matrices) convention.